projects/igniteui-angular/src/lib/services/transaction/transaction.ts

```diff
@@ -1,11 +1,20 @@
 import { EventEmitter } from '@angular/core';
+import { ThrowStmt } from '@angular/compiler';
 
 export enum TransactionType {
     ADD = 'add',
     DELETE = 'delete',
     UPDATE = 'update'
 }
 
+export enum TransactionEventOrigin {
```

Comment:
@wnvko
Looking at the listed event origins for onStateUpdate I don't see when it is emitted from a commit. Looking at the IgxTransactionService I also didn't notice any place on commit that the onStateUpdate is called except on clear. Is this the expected behavior of the onStateUpdate event?

Also due to the IgxHierarchicalTransactionService commit calling clear() a second time when childDataKey is not provided, the onStateUpdate is emitted two times which I also am not sure if it's a valid scenario anyway in the igxTreeGrid but just asking.

Reply:
@skrustev 
Calling `onStateUpdate` in `clear` is OK from my point of view.
Calling `clear` in `IgxHierarchicalTransactionService.commit` is a bug. You can log this, as I do not thing it is OK to fix it in this feature PR.

Reply:
@wnvko 
Ok so if the user wants to perform action on `commit` he will have to check for the `clear` event origin ? It's kinda weird to me but if you think it's ok I am fine as well.

Reply:
@skrustev @wnvko I have logged a separate issue (#7615) to address that problem.